frame/staking/src/voter_bags/thresholds.rs

```diff
@@ -0,0 +1,229 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Generated voter bag thresholds.
+
+use super::N_BAGS;
+
+/// Ratio between adjacent bags;
+#[cfg(any(test, feature = "std"))]
+#[allow(unused)]
+pub const CONSTANT_RATIO: f64 = 1.2483305489016119;
+
+/// Upper thresholds for each bag.
+pub const THRESHOLDS: [u64; N_BAGS as usize] = [
```

Comment:
looks like something for a macro

Reply:
Yes, but I couldn't figure out a good way to generate it with a declarative macro, and didn't want to go to the trouble of writing a proc-macro for it. Instead, wrote the generation script [here](https://github.com/paritytech/substrate/blob/92220ec6213f609dee9c528f501e8d7f73e4e4bd/frame/staking/src/bin/make_bags.rs). I can convert that to a proc-macro if you like, but IMO it's not really worth it.